cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletEndpointTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet;
+
+import com.linkedin.kafka.cruisecontrol.async.OperationFuture;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.CruiseControlParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterUtils;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.UserTasksParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.response.CruiseControlResponse;
+import com.linkedin.kafka.cruisecontrol.servlet.response.UserTaskState;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import kafka.utils.MockTime;
+import org.apache.kafka.common.utils.Time;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class KafkaCruiseControlServletEndpointTest {
+  private static final String GET_METHOD = "GET";
+  private static final String POST_METHOD = "POST";
+  private static final Function<String, OperationFuture> FUTURE_CREATOR = uuid -> new OperationFuture("future");
+
+  // A hack to make 2 requests to same endpoint 'look' different to UserTaskManager
+  private static final Map<String, String[]> DIFF_PARAM = new HashMap<>();
+  private static final Map<String, String[]> EMPTY_PARAM = Collections.emptyMap();
+
+  private static final UUID repeatUUID = UUID.randomUUID();
+
+  private static Collection<Object[]> _initializeServletRequestsOutput = new ArrayList<>();
+  private static Collection<Object[]> _populateUserTaskManagerOutput = new ArrayList<>();
+  private static Time _mockTime;
```

Comment:
One functions sets value to the variables and another function get values from the variables, it's easier to make them static to avoid passing them as arguments.